Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -2585,15 +2678,48 @@ bool Parser::match_statement() const
 bool Parser::match_declaration() const
 {
     auto type = m_state.current_token.type();
+
+    if (type == TokenType::Let && !m_state.strict_mode) {
+        return try_match_let_declaration();
+    }
+
     return type == TokenType::Function
         || type == TokenType::Class
         || type == TokenType::Const
         || type == TokenType::Let;
 }
 
+bool Parser::try_match_let_declaration() const
+{
+    VERIFY(m_state.current_token.type() == TokenType::Let);
+
+    auto& mutable_me = const_cast<Parser&>(*this);
+
+    mutable_me.save_state();
+
+    ScopeGuard state_rollback = [&] {
+        mutable_me.load_state();
+    };
+
+    mutable_me.consume(TokenType::Let);
```

Comment:
Using `const_cast` is not great but we don't have peek so this was the best way I could think of.
One alternative is creating a full new lexer starting at the let.

Reply:
just don't mark the method `const` if it isn't?